src/components/PostCard/PostHeader/PostHeader.js

```diff
@@ -114,71 +117,93 @@ export default class PostHeader extends PureComponent {
     }
 
     const showNormal = ((canBeCompleted && canEdit && expanded) && (topics?.length > 0 || (canHaveTimes && timeWindow.length > 0))) || false
-
-    return <div styleName={cx('header', { constrained }, { detailHasImage })} className={className}>
-      <div styleName='headerMainRow'>
-        <div styleName='headerTopRow'>
-          <Avatar avatarUrl={creator.avatarUrl} url={creatorUrl} styleName='avatar' />
-          <div styleName='headerText'>
-            <Highlight {...highlightProps}>
-              <Link to={creatorUrl} styleName='userName' data-tip={creator.tagline} data-for='announcement-tt'>{creator.name}</Link>
-            </Highlight>
-            <div styleName='timestampRow'>
-              <span styleName='timestamp'>
-                {TextHelpers.humanDate(createdAt)}
-              </span>
-              {announcement && <span styleName='announcementSection'>
-                <span styleName='announcementSpacer'>•</span>
-                <span data-tip='Announcement' data-for='announcement-tt'>
-                  <Icon name='Announcement' styleName='announcementIcon' />
+    const currentGroup = groups.find(group => group.slug === routeParams.groupSlug)
```

Comment:
I don't think it will be necessary for the amount of compute that is occurring

https://beta.reactjs.org/reference/react/useMemo

![image](https://user-images.githubusercontent.com/5193218/221301643-97afd6af-b847-4f1a-910d-bc244c78d00e.png)